addons/hr_recruitment/models/hr_applicant.py

```diff
@@ -1027,7 +1027,7 @@ def create_employee_from_applicant(self):
         action['res_id'] = employee.id
         employee_attachments = self.env['ir.attachment'].search([('res_model', '=','hr.employee'), ('res_id', '=', employee.id)])
         unique_attachments = self.attachment_ids.filtered(
-            lambda attachment: attachment.datas not in employee_attachments.mapped('datas')
+            lambda attachment: attachment.raw not in employee_attachments.mapped('raw')
```

Comment:
This should at least use the `checksum` not read all the attachment data from disk.